api/CMakeLists.txt

```diff
@@ -98,6 +98,16 @@ if(WITH_ASYNC_EXPORT_PREVIEW)
   target_compile_definitions(opentelemetry_api INTERFACE ENABLE_ASYNC_EXPORT)
 endif()
 
+# A better place should be in sdk, not api
+if(WITH_OTLP_HTTP_SSL_PREVIEW)
+  target_compile_definitions(opentelemetry_api INTERFACE ENABLE_OTLP_HTTP_SSL)
+
+  if(WITH_OTLP_HTTP_SSL_TLS_PREVIEW)
+    target_compile_definitions(opentelemetry_api
+                               INTERFACE ENABLE_OTLP_HTTP_SSL_TLS)
```

Comment:
nit - should we keep the macro name also prefixed with `_PREVIEW`, as per the existing release guidelines - https://github.com/open-telemetry/opentelemetry-cpp/blob/main/Versioning.md#release-policy.

Reply:
Agreed.

Fixed both feature flags to have a _PREVIEW suffix.